sos/report/plugins/ceph_rgw.py

```diff
@@ -57,4 +81,52 @@ def setup(self):
                 "/etc/ceph/*bindpass*"
             ])
 
+        # Get commands output for both Ceph and microCeph
+        rgw_id = "radosgw.gateway" if microceph else "rgw." + gethostname()
+        self.add_cmd_output([f"radosgw-admin --id={rgw_id} {c}" for c in cmds])
+
+        # Get all the zone data
+        res = self.collect_cmd_output(f'radosgw-admin --id={rgw_id} zone list')
+        if res['status'] == 0:
+            try:
+                _out = json.loads(res['output'])
+                zone_list = _out['zones']
+                for zone in zone_list:
+                    _cmd = (
+                        f'radosgw-admin --id={rgw_id} '
+                        f'zone get --rgw-zone={zone}'
+                    )
+                    self.add_cmd_output(_cmd)
+                    self.cmd_list.append(_cmd)
+            except ValueError as err:
+                self._log_error(f'Error while getting get rgw '
+                                f'zone list: {err}')
+
+        # Get all the zonegroup data
+        res = self.collect_cmd_output(f'radosgw-admin --id={rgw_id} '
+                                      f'zonegroup list')
+        if res['status'] == 0:
+            try:
+                _out = json.loads(res['output'])
+                zonegroups = _out['zonegroups']
+                for zgroup in zonegroups:
+                    _cmd = (
+                        f'radosgw-admin --id={rgw_id} '
+                        f'zone get --rgw-zonegroup={zgroup}'
+                    )
+                    self.add_cmd_output(_cmd)
+                    self.cmd_list.append(_cmd)
+            except ValueError as err:
+                self._log_error(f'Error while getting get rgw '
+                                f'zonegroup list: {err}')
+
+    def postproc(self):
+        """ Obfuscate secondary zone access keys """
+
+        rsub = r'("access_key":|"secret_key":)\s.*'
+
+        for _cmd in self.cmd_list:
+            self.do_cmd_output_sub(_cmd, rsub, r'\1 "**********"')
```

Comment:
You don't actually need to know the list of commands to do this. You could just do `radosgw-admin` as a text to the `do_cmd_output_sub`, and it should sort itself out.